### ISSUE TITLE
Intermittent Bounds Error

### ISSUE THREAD

**Issue by lstagner** [JuliaStats/Lasso.jl#42]
Sometimes when solving a model or more commonly when doing cross validation I get a BoundsError
```
m = selectmodel(path, MinCV1se(path));
BoundsError: attempt to access 306-element Array{Float64,1} at index [307]

Stacktrace:
 [1] getindex at ./array.jl:744 [inlined]
 [2] update_coef! at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:189 [inlined]
 [3] cycle!(::Lasso.SparseCoefficients{Float64}, ::NaiveCoordinateDescent{Float64,true,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}, ::Float64, ::Bool) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:246
 [4] cdfit!(::Lasso.SparseCoefficients{Float64}, ::NaiveCoordinateDescent{Float64,true,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}, ::Float64, ::Symbol) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:598
 [5] #fit!#14(::Bool, ::Int64, ::Float64, ::Float64, ::Bool, ::Symbol, ::Float64, ::typeof(fit!), ::LassoPath{LinearModel{GLM.LmResp{Array{Float64,1}},NaiveCoordinateDescent{Float64,true,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}},Float64}) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:827
 [6] (::StatsBase.var"#kw##fit!")(::NamedTuple{(:irls_tol,),Tuple{Float64}}, ::typeof(fit!), ::LassoPath{LinearModel{GLM.LmResp{Array{Float64,1}},NaiveCoordinateDescent{Float64,true,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}},Float64}) at ./none:0
 [7] #fit#3(::Array{Float64,1}, ::Array{Float64,1}, ::Float64, ::Int64, ::Float64, ::Array{Float64,1}, ::Bool, ::Bool, ::Type, ::Bool, ::Float64, ::Bool, ::Int64, ::Nothing, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{,Tuple{}}}, ::typeof(fit), ::Type{LassoPath}, ::Array{Float64,2}, ::Array{Float64,1}, ::Normal{Float64}, ::IdentityLink) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/Lasso.jl:472
 [8] (::StatsBase.var"#kw##fit")(::NamedTuple{(:λ, :wts, :offset, :standardize),Tuple{Array{Float64,1},Array{Float64,1},Array{Float64,1},Bool}}, ::typeof(fit), ::Type{LassoPath}, ::Array{Float64,2}, ::Array{Float64,1}, ::Normal{Float64}, ::IdentityLink) at ./none:0
 [9] #cross_validate_path#48(::Array{Float64,1}, ::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:standardize,),Tuple{Bool}}}, ::typeof(cross_validate_path), ::LassoPath{LinearModel{GLM.LmResp{Array{Float64,1}},NaiveCoordinateDescent{Float64,false,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}},Float64}, ::Array{Float64,2}, ::Array{Float64,1}, ::MinCV1se) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/cross_validation.jl:73
 [10] #cross_validate_path at ./none:0 [inlined]
 [11] #cross_validate_path#49 at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/cross_validation.jl:113 [inlined]
 [12] cross_validate_path at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/cross_validation.jl:109 [inlined]
 [13] segselect at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/segselect.jl:109 [inlined]
 [14] coef at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/segselect.jl:86 [inlined]
 [15] selectmodel(::LassoPath{LinearModel{GLM.LmResp{Array{Float64,1}},NaiveCoordinateDescent{Float64,false,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}},Float64}, ::MinCV1se) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/segselect.jl:191
 [16] top-level scope at In[1355]:1
```
The bounds error seems to occur randomly which given how cross validation works makes sense.

**Comment by lstagner** [comment]
Here is a MWE to help debug
[lasso_boundserror.jld2.zip](https://github.com/JuliaStats/Lasso.jl/files/4209003/lasso_boundserror.jld2.zip)
```
using Lasso, JLD2
@load "lasso_boundserror.jld2"

fit(LassoPath, A, b; intercept=false);
BoundsError: attempt to access 340-element Array{Float64,1} at index [341]

Stacktrace:
 [1] update_coef! at ./array.jl:744 [inlined]
 [2] cycle!(::Lasso.SparseCoefficients{Float64}, ::NaiveCoordinateDescent{Float64,false,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}, ::Float64, ::Bool) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:246
 [3] cdfit!(::Lasso.SparseCoefficients{Float64}, ::NaiveCoordinateDescent{Float64,false,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}, ::Float64, ::Symbol) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:598
 [4] #fit!#14(::Bool, ::Int64, ::Float64, ::Float64, ::Bool, ::Symbol, ::Float64, ::typeof(fit!), ::LassoPath{LinearModel{GLM.LmResp{Array{Float64,1}},NaiveCoordinateDescent{Float64,false,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}},Float64}) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/coordinate_descent.jl:827
 [5] (::StatsBase.var"#kw##fit!")(::NamedTuple{(:irls_tol,),Tuple{Float64}}, ::typeof(fit!), ::LassoPath{LinearModel{GLM.LmResp{Array{Float64,1}},NaiveCoordinateDescent{Float64,false,Array{Float64,2},Lasso.RandomCoefficientIterator,Nothing}},Float64}) at ./none:0
 [6] #fit#3(::Array{Float64,1}, ::Array{Float64,1}, ::Float64, ::Int64, ::Float64, ::Nothing, ::Bool, ::Bool, ::Type, ::Bool, ::Float64, ::Bool, ::Int64, ::Nothing, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{,Tuple{}}}, ::typeof(fit), ::Type{LassoPath}, ::Array{Float64,2}, ::Array{Float64,1}, ::Normal{Float64}, ::IdentityLink) at /Users/lstagner/.julia/packages/Lasso/7IQvf/src/Lasso.jl:472
 [7] (::StatsBase.var"#kw##fit")(::NamedTuple{(:intercept,),Tuple{Bool}}, ::typeof(fit), ::Type{LassoPath}, ::Array{Float64,2}, ::Array{Float64,1}, ::Normal{Float64}, ::IdentityLink) at ./none:0 (repeats 3 times)
 [8] top-level scope at In[1502]:2
```

**Comment by lstagner** [comment]
So it looks like the bounds error happens when `maxncoef` is exceeded. A more informative error message would be helpful. Making it larger partially fixes the issue. Now though it seems like cross validation is not respecting user set `maxncoef` and is encountering the same bounds error.

**Comment by AsafManela** [comment]
Fixed by #44
